### PR TITLE
Validate distributin channels and systems

### PR DESCRIPTION
**Pull request by elmgren**
I'll write new test in another branch when these changes are approved

### CODE REVIEW

File: data-catalog-backend-app/src/main/java/no/nav/data/catalog/backend/app/system/SystemRepository.java [hunk 1]
```diff
@@ -2,11 +2,16 @@
...
 
+import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
 
 public interface SystemRepository extends JpaRepository<System, UUID>, JpaSpecificationExecutor<System> {
 	Optional<System> findByName(@Param("name") String name);
+
+	@Query(value = "select * from system where json_property->>'name' = (?1)", nativeQuery = true)
+	List<System> findAllByName(List<String> name);
```
Comment (skotsj) [inline data-catalog-backend-app/src/main/java/no/nav/data/catalog/backend/app/system/SystemRepository.java]: system har vel ikke jsonfelt?

File: data-catalog-backend-app/src/main/java/no/nav/data/catalog/backend/app/distributionchannel/DistributionChannelService.java [hunk 2]
```diff
@@ -95,18 +128,48 @@ private void attachSystems(DistributionChannelRequest request, DistributionChann
...
+		if (requests.isEmpty()) {
+			return Collections.emptyList();
+		}
+
+		List<ValidationError> validationErrors = new ArrayList<>(validateNoDuplicates(requests));
+		validationErrors.addAll(validateDistributionChannelRequests(requests));
+		return validationErrors;
+	}
+
+	private List<ValidationError> validateDistributionChannelRequests(List<DistributionChannelRequest> requests) {
```
Comment (skotsj) [inline data-catalog-backend-app/src/main/java/no/nav/data/catalog/backend/app/distributionchannel/DistributionChannelService.java]: fiks metodenavn/oppdeling

File: data-catalog-backend-app/src/main/java/no/nav/data/catalog/backend/app/system/SystemRepository.java (hunk 1 above)
Comment (elmgren) [reply data-catalog-backend-app/src/main/java/no/nav/data/catalog/backend/app/system/SystemRepository.java]: Fixed. Burde stå outdated her.

File: data-catalog-backend-app/src/main/java/no/nav/data/catalog/backend/app/distributionchannel/DistributionChannelService.java (hunk 2 above)
Comment (elmgren) [reply data-catalog-backend-app/src/main/java/no/nav/data/catalog/backend/app/distributionchannel/DistributionChannelService.java]: Fixed

**Pull request merged**